diff --git a/vendor/brotli/c/include/brotli/decode.h b/vendor/brotli/c/include/brotli/decode.h
index 8d436db733..31a0d7290c 100644
--- a/vendor/brotli/c/include/brotli/decode.h
+++ b/vendor/brotli/c/include/brotli/decode.h
@@ -12,9 +12,9 @@
 #ifndef BROTLI_DEC_DECODE_H_
 #define BROTLI_DEC_DECODE_H_
 
-#include <brotli/port.h>
-#include <brotli/shared_dictionary.h>
-#include <brotli/types.h>  /* IWYU pragma: export */
+#include "port.h"
+#include "shared_dictionary.h"
+#include "types.h"  /* IWYU pragma: export */
 
 #if defined(__cplusplus) || defined(c_plusplus)
 extern "C" {
diff --git a/vendor/brotli/c/include/brotli/encode.h b/vendor/brotli/c/include/brotli/encode.h
index ac26ea8e62..483904b37c 100644
--- a/vendor/brotli/c/include/brotli/encode.h
+++ b/vendor/brotli/c/include/brotli/encode.h
@@ -12,9 +12,9 @@
 #ifndef BROTLI_ENC_ENCODE_H_
 #define BROTLI_ENC_ENCODE_H_
 
-#include <brotli/port.h>
-#include <brotli/shared_dictionary.h>
-#include <brotli/types.h>  /* IWYU pragma: export */
+#include "port.h"
+#include "shared_dictionary.h"
+#include "types.h"  /* IWYU pragma: export */
 
 #if defined(__cplusplus) || defined(c_plusplus)
 extern "C" {
diff --git a/vendor/brotli/c/include/brotli/shared_dictionary.h b/vendor/brotli/c/include/brotli/shared_dictionary.h
index 2970c2dcc9..17d772f313 100644
--- a/vendor/brotli/c/include/brotli/shared_dictionary.h
+++ b/vendor/brotli/c/include/brotli/shared_dictionary.h
@@ -9,8 +9,8 @@
 #ifndef BROTLI_COMMON_SHARED_DICTIONARY_H_
 #define BROTLI_COMMON_SHARED_DICTIONARY_H_
 
-#include <brotli/port.h>
-#include <brotli/types.h>
+#include "port.h"
+#include "types.h"
 
 #if defined(__cplusplus) || defined(c_plusplus)
 extern "C" {
