diff --git a/vendor/brotli/c/common/platform.h b/vendor/brotli/c/common/platform.h
index 7406f3fe6..e3f8b05dd 100644
--- a/vendor/brotli/c/common/platform.h
+++ b/vendor/brotli/c/common/platform.h
@@ -272,6 +272,8 @@ OR:
 
 #if defined(BROTLI_BUILD_NO_UNALIGNED_READ_FAST)
 #define BROTLI_UNALIGNED_READ_FAST (!!0)
+#elif !defined(NDEBUG)
+#define BROTLI_UNALIGNED_READ_FAST (!!0)
 #elif defined(BROTLI_TARGET_X86) || defined(BROTLI_TARGET_X64) ||       \
     defined(BROTLI_TARGET_ARMV7) || defined(BROTLI_TARGET_ARMV8_ANY) || \
     defined(BROTLI_TARGET_RISCV64) || defined(BROTLI_TARGET_LOONGARCH64)
