diff --git a/vendor/brotli/c/include/brotli/decode.h b/vendor/brotli/c/include/brotli/decode.h
index af1aa23f6..3fd646a3d 100644
--- a/vendor/brotli/c/include/brotli/decode.h
+++ b/vendor/brotli/c/include/brotli/decode.h
@@ -12,9 +12,9 @@
 #ifndef BROTLI_DEC_DECODE_H_
 #define BROTLI_DEC_DECODE_H_
 
-#include <brotli/port.h>
-#include <brotli/shared_dictionary.h>
-#include <brotli/types.h>
+#include "port.h"
+#include "shared_dictionary.h"
+#include "types.h"
 
 #if defined(__cplusplus) || defined(c_plusplus)
 extern "C" {
diff --git a/vendor/brotli/c/include/brotli/encode.h b/vendor/brotli/c/include/brotli/encode.h
index dea9931eb..a31bb328a 100644
--- a/vendor/brotli/c/include/brotli/encode.h
+++ b/vendor/brotli/c/include/brotli/encode.h
@@ -12,9 +12,9 @@
 #ifndef BROTLI_ENC_ENCODE_H_
 #define BROTLI_ENC_ENCODE_H_
 
-#include <brotli/port.h>
-#include <brotli/shared_dictionary.h>
-#include <brotli/types.h>
+#include "port.h"
+#include "shared_dictionary.h"
+#include "types.h"
 
 #if defined(__cplusplus) || defined(c_plusplus)
 extern "C" {
