diff --git a/vendor/c-ares/src/lib/ares_options.c b/vendor/c-ares/src/lib/ares_options.c
index 3082f33245..8159461423 100644
--- a/vendor/c-ares/src/lib/ares_options.c
+++ b/vendor/c-ares/src/lib/ares_options.c
@@ -455,15 +455,9 @@ ares_status_t ares_init_by_options(ares_channel_t            *channel,
     }
   }
 
-  /* As of c-ares 1.31.0, the Query Cache is on by default.  The only way to
-   * disable it is to set options->qcache_max_ttl = 0 while specifying the
-   * ARES_OPT_QUERY_CACHE which will actually disable it completely. */
   if (optmask & ARES_OPT_QUERY_CACHE) {
     /* qcache_max_ttl is unsigned unlike the others */
     channel->qcache_max_ttl = options->qcache_max_ttl;
-  } else {
-    optmask                 |= ARES_OPT_QUERY_CACHE;
-    channel->qcache_max_ttl  = 3600;
   }
 
   /* Initialize the ipv4 servers if provided */
