diff --git a/vendor/cmark-gfm/extensions/cmark-gfm-core-extensions.h b/vendor/cmark-gfm/extensions/cmark-gfm-core-extensions.h
index bc29ffd41..406355fdc 100644
--- a/vendor/cmark-gfm/extensions/cmark-gfm-core-extensions.h
+++ b/vendor/cmark-gfm/extensions/cmark-gfm-core-extensions.h
@@ -5,8 +5,8 @@
 extern "C" {
 #endif
 
-#include "cmark-gfm-extension_api.h"
-#include "cmark-gfm_export.h"
+#include "../src/cmark-gfm-extension_api.h"
+#include "../src/cmark-gfm_export.h"
 #include <stdbool.h>
 #include <stdint.h>
 
