diff --git a/vendor/dragonbox/include/dragonbox/dragonbox.h b/vendor/dragonbox/include/dragonbox/dragonbox.h
index a7446686d..10cf8e92d 100644
--- a/vendor/dragonbox/include/dragonbox/dragonbox.h
+++ b/vendor/dragonbox/include/dragonbox/dragonbox.h
@@ -634,7 +634,7 @@ namespace jkj::dragonbox {
 				for (int i = 1; i < bit_width; ++i) {
 					mod_inverse = mod_inverse * mod_inverse * a;
 				}
-				if (bit_width < value_bits<UInt>) {
+				if (bit_width < int(value_bits<UInt>)) {
 					auto mask = UInt((UInt(1) << bit_width) - 1);
 					return UInt(mod_inverse & mask);
 				}
