diff --git b/vendor/libfuse/include/fuse_config.h a/vendor/libfuse/include/fuse_config.h
new file mode 100644
index 0000000000..fd931af62e
--- /dev/null
+++ a/vendor/libfuse/include/fuse_config.h
@@ -0,0 +1,59 @@
+#pragma once
+
+#define PACKAGE_VERSION "3.17.4"
+
+#if defined(__linux__)
+    #define HAVE_CLOSE_RANGE
+    #define HAVE_COPY_FILE_RANGE
+    #define HAVE_FALLOCATE
+    #define HAVE_FDATASYNC
+    #define HAVE_FORK
+    #define HAVE_FSTATAT
+    // #define HAVE_ICONV
+    #define HAVE_OPENAT
+    #define HAVE_PIPE2
+    #define HAVE_POSIX_FALLOCATE
+    #define HAVE_READLINKAT
+    #define HAVE_SETXATTR
+    #define HAVE_SPLICE
+    #define HAVE_STRUCT_STAT_ST_ATIM
+    // #define HAVE_STRUCT_STAT_ST_ATIMESPEC
+    #define HAVE_UTIMENSAT
+    #define HAVE_VMSPLICE
+#elif defined(__FreeBSD__)
+    // #define HAVE_CLOSE_RANGE
+    #define HAVE_COPY_FILE_RANGE
+    // #define HAVE_FALLOCATE
+    #define HAVE_FDATASYNC
+    #define HAVE_FORK
+    #define HAVE_FSTATAT
+    // #define HAVE_ICONV
+    #define HAVE_OPENAT
+    #define HAVE_PIPE2
+    #define HAVE_POSIX_FALLOCATE
+    #define HAVE_READLINKAT
+    // #define HAVE_SETXATTR
+    // #define HAVE_SPLICE
+    #define HAVE_STRUCT_STAT_ST_ATIM
+    #define HAVE_STRUCT_STAT_ST_ATIMESPEC
+    #define HAVE_UTIMENSAT
+    // #define HAVE_VMSPLICE
+#elif defined(__OpenBSD__)
+    // #define HAVE_CLOSE_RANGE
+    // #define HAVE_COPY_FILE_RANGE
+    // #define HAVE_FALLOCATE
+    #define HAVE_FDATASYNC
+    #define HAVE_FORK
+    #define HAVE_FSTATAT
+    // #define HAVE_ICONV
+    #define HAVE_OPENAT
+    #define HAVE_PIPE2
+    // #define HAVE_POSIX_FALLOCATE
+    #define HAVE_READLINKAT
+    // #define HAVE_SETXATTR
+    // #define HAVE_SPLICE
+    #define HAVE_STRUCT_STAT_ST_ATIM
+    #define HAVE_STRUCT_STAT_ST_ATIMESPEC
+    #define HAVE_UTIMENSAT
+    // #define HAVE_VMSPLICE
+#endif
diff --git b/vendor/libfuse/include/libfuse_config.h a/vendor/libfuse/include/libfuse_config.h
new file mode 100644
index 0000000000..2da5803eaf
--- /dev/null
+++ a/vendor/libfuse/include/libfuse_config.h
@@ -0,0 +1,11 @@
+#pragma once
+
+#define FUSE_HOTFIX_VERSION 4
+
+#define FUSE_MAJOR_VERSION 3
+
+#define FUSE_MINOR_VERSION 17
+
+#define FUSE_RC_VERSION
+
+// #define LIBFUSE_BUILT_WITH_VERSIONED_SYMBOLS 1
