diff --git b/vendor/libmicrohttpd/src/include/microhttpd.h a/vendor/libmicrohttpd/src/include/microhttpd.h
index 1d62f2ccf..06536f2c7 100644
--- b/vendor/libmicrohttpd/src/include/microhttpd.h
+++ a/vendor/libmicrohttpd/src/include/microhttpd.h
@@ -3709,6 +3709,18 @@ MHD_create_response_from_iovec (const struct MHD_IoVec *iov,
                                 void *cls);
 
 
+/**
+ * Move response headers from one response object to another.
+ *
+ * @param src response object to steal from
+ * @param dest response object to move headers to
+ * @ingroup response
+ */
+_MHD_EXTERN void
+MHD_move_response_headers (struct MHD_Response *src,
+                           struct MHD_Response *dest);
+
+
 /**
  * Enumeration for actions MHD should perform on the underlying socket
  * of the upgrade.  This API is not finalized, and in particular
diff --git b/vendor/libmicrohttpd/src/microhttpd/response.c a/vendor/libmicrohttpd/src/microhttpd/response.c
index ca3639f49..ed3da6378 100644
--- b/vendor/libmicrohttpd/src/microhttpd/response.c
+++ a/vendor/libmicrohttpd/src/microhttpd/response.c
@@ -1536,6 +1536,36 @@ MHD_create_response_from_iovec (const struct MHD_IoVec *iov,
 }
 
 
+/**
+ * Move response headers from one response object to another.
+ *
+ * @param src response object to steal from
+ * @param dest response object to move headers to
+ * @ingroup response
+ */
+_MHD_EXTERN void
+MHD_move_response_headers (struct MHD_Response *src,
+                           struct MHD_Response *dest)
+{
+  struct MHD_HTTP_Header *last_header;
+
+  if (NULL == src->first_header)
+    return;
+
+  if (NULL == dest->first_header) {
+    dest->first_header = src->first_header;
+    dest->last_header = src->last_header;
+  } else {
+    dest->last_header->next = src->first_header;
+    src->first_header->prev = dest->last_header;
+    dest->last_header = src->last_header;
+  }
+
+  src->first_header = NULL;
+  src->last_header = NULL;
+}
+
+
 #ifdef UPGRADE_SUPPORT
 /**
  * This connection-specific callback is provided by MHD to
