diff --git a/vendor/libssh/src/buffer.c b/vendor/libssh/src/buffer.c
index 449fa941c7..e00b094f32 100644
--- a/vendor/libssh/src/buffer.c
+++ b/vendor/libssh/src/buffer.c
@@ -28,7 +28,9 @@
 #include <stdbool.h>
 #include <stdio.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/legacy.c b/vendor/libssh/src/legacy.c
index 65a47d6e63..601a8e149d 100644
--- a/vendor/libssh/src/legacy.c
+++ b/vendor/libssh/src/legacy.c
@@ -26,6 +26,9 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <errno.h>
 #include <stdio.h>
 
diff --git a/vendor/libssh/src/messages.c b/vendor/libssh/src/messages.c
index a9f5279f8a..3fa98dd329 100644
--- a/vendor/libssh/src/messages.c
+++ b/vendor/libssh/src/messages.c
@@ -26,7 +26,9 @@
 #include <string.h>
 #include <stdlib.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/packet_cb.c b/vendor/libssh/src/packet_cb.c
index 54c94c8438..ca54191a0b 100644
--- a/vendor/libssh/src/packet_cb.c
+++ b/vendor/libssh/src/packet_cb.c
@@ -24,6 +24,9 @@
 #include "config.h"
 
 #include <stdlib.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #ifdef HAVE_ARPA_INET_H
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/packet_crypt.c b/vendor/libssh/src/packet_crypt.c
index 96e9586c02..ad9ca4238a 100644
--- a/vendor/libssh/src/packet_crypt.c
+++ b/vendor/libssh/src/packet_crypt.c
@@ -27,7 +27,9 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/pki.c b/vendor/libssh/src/pki.c
index a719d85f70..28eff44b44 100644
--- a/vendor/libssh/src/pki.c
+++ b/vendor/libssh/src/pki.c
@@ -35,6 +35,9 @@
 #include "config.h"
 #include "libssh/wrapper.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <errno.h>
 #include <ctype.h>
 #include <stdint.h>
diff --git a/vendor/libssh/src/sftp_aio.c b/vendor/libssh/src/sftp_aio.c
index c1c54561ae..37eb5a6bad 100644
--- a/vendor/libssh/src/sftp_aio.c
+++ b/vendor/libssh/src/sftp_aio.c
@@ -25,6 +25,10 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
+
 #include "libssh/sftp.h"
 #include "libssh/sftp_priv.h"
 #include "libssh/buffer.h"
diff --git a/vendor/libssh/src/sftp_common.c b/vendor/libssh/src/sftp_common.c
index 13512b8dbe..1fab7d1904 100644
--- a/vendor/libssh/src/sftp_common.c
+++ b/vendor/libssh/src/sftp_common.c
@@ -26,6 +26,9 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <ctype.h>
 
 #include "libssh/sftp.h"
diff --git a/vendor/libssh/src/string.c b/vendor/libssh/src/string.c
index 0ab9310c4e..e1e63fd0ac 100644
--- a/vendor/libssh/src/string.c
+++ b/vendor/libssh/src/string.c
@@ -26,7 +26,9 @@
 #include <errno.h>
 #include <limits.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/ttyopts.c b/vendor/libssh/src/ttyopts.c
index a1ab31f0c7..88ff1581a8 100644
--- a/vendor/libssh/src/ttyopts.c
+++ b/vendor/libssh/src/ttyopts.c
@@ -30,6 +30,9 @@
 #include <libssh/priv.h>
 #include <string.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #ifdef HAVE_TERMIOS_H
 #include <termios.h>
 #endif
