diff --git a/vendor/libssh/src/threads/mbedtls.c b/vendor/libssh/src/threads/mbedtls.c
index 62e1db8756..77268cda56 100644
--- a/vendor/libssh/src/threads/mbedtls.c
+++ b/vendor/libssh/src/threads/mbedtls.c
@@ -51,13 +51,15 @@ int crypto_thread_init(struct ssh_threads_callbacks_struct *user_callbacks)
     }
 #elif defined MBEDTLS_THREADING_PTHREAD
     return SSH_OK;
+#elif defined MBEDTLS_THREADING_WIN32
+    return SSH_OK;
 #else
     fprintf(stderr,
             "MbedTLS needs to have threading enabled with "
-            "MBEDTLS_THREADING_PTHREAD or MBEDTLS_THREADING_ALT "
+            "MBEDTLS_THREADING_PTHREAD or MBEDTLS_THREADING_WIN32 or MBEDTLS_THREADING_ALT "
             "in mbedtls_config.h\n");
 #warn "MbedTLS needs to have threading enabled with " \
-      "MBEDTLS_THREADING_PTHREAD or MBEDTLS_THREADING_ALT in mbedtls_config.h"
+      "MBEDTLS_THREADING_PTHREAD or MBEDTLS_THREADING_WIN32 or MBEDTLS_THREADING_ALT in mbedtls_config.h"
     return SSH_ERROR;
 #endif
 }
