diff --git b/vendor/libssh/include/libssh/agent.h a/vendor/libssh/include/libssh/agent.h
index 466a884fc..720521596 100644
--- b/vendor/libssh/include/libssh/agent.h
+++ a/vendor/libssh/include/libssh/agent.h
@@ -21,7 +21,7 @@
 #ifndef __AGENT_H
 #define __AGENT_H
 
-#include "libssh.h"
+#include "libssh/libssh.h"
 
 /* Messages for the authentication agent connection. */
 #define SSH_AGENTC_REQUEST_RSA_IDENTITIES        1
diff --git b/vendor/libssh/include/libssh/auth.h a/vendor/libssh/include/libssh/auth.h
index ee7a36bd1..90b377d4f 100644
--- b/vendor/libssh/include/libssh/auth.h
+++ a/vendor/libssh/include/libssh/auth.h
@@ -21,7 +21,7 @@
 #ifndef AUTH_H_
 #define AUTH_H_
 #include "config.h"
-#include "callbacks.h"
+#include "libssh/callbacks.h"
 
 SSH_PACKET_CALLBACK(ssh_packet_userauth_banner);
 SSH_PACKET_CALLBACK(ssh_packet_userauth_failure);
diff --git b/vendor/libssh/include/libssh/bignum.h a/vendor/libssh/include/libssh/bignum.h
index 328fbc99a..726ed7b9a 100644
--- b/vendor/libssh/include/libssh/bignum.h
+++ a/vendor/libssh/include/libssh/bignum.h
@@ -21,9 +21,9 @@
 #ifndef BIGNUM_H_
 #define BIGNUM_H_
 
-#include "libcrypto.h"
-#include "libgcrypt.h"
-#include "libmbedcrypto.h"
+#include "libssh/libcrypto.h"
+#include "libssh/libgcrypt.h"
+#include "libssh/libmbedcrypto.h"
 
 bignum ssh_make_string_bn(ssh_string string);
 ssh_string ssh_make_bignum_string(bignum num);
diff --git b/vendor/libssh/include/libssh/bind.h a/vendor/libssh/include/libssh/bind.h
index 9035307a2..c0439d2c1 100644
--- b/vendor/libssh/include/libssh/bind.h
+++ a/vendor/libssh/include/libssh/bind.h
@@ -21,9 +21,9 @@
 #ifndef BIND_H_
 #define BIND_H_
 
-#include "priv.h"
-#include "kex.h"
-#include "session.h"
+#include "libssh/priv.h"
+#include "libssh/kex.h"
+#include "libssh/session.h"
 
 struct ssh_bind_struct {
   struct ssh_common_struct common; /* stuff common to ssh_bind and ssh_session */
diff --git b/vendor/libssh/include/libssh/bind_config.h a/vendor/libssh/include/libssh/bind_config.h
index 39fb719c8..7ee19b873 100644
--- b/vendor/libssh/include/libssh/bind_config.h
+++ a/vendor/libssh/include/libssh/bind_config.h
@@ -26,7 +26,7 @@
 #ifndef BIND_CONFIG_H_
 #define BIND_CONFIG_H_
 
-#include "server.h"
+#include "libssh/server.h"
 
 enum ssh_bind_config_opcode_e {
     /* Known but not allowed in Match block */
diff --git b/vendor/libssh/include/libssh/buffer.h a/vendor/libssh/include/libssh/buffer.h
index 7f0085347..a55a1b402 100644
--- b/vendor/libssh/include/libssh/buffer.h
+++ a/vendor/libssh/include/libssh/buffer.h
@@ -23,7 +23,7 @@
 
 #include <stdarg.h>
 
-#include "libssh.h"
+#include "libssh/libssh.h"
 
 #define SSH_BUFFER_PACK_END ((uint32_t) 0x4f65feb3)
 
diff --git b/vendor/libssh/include/libssh/callbacks.h a/vendor/libssh/include/libssh/callbacks.h
index 9e5f2409e..36fe7f8c0 100644
--- b/vendor/libssh/include/libssh/callbacks.h
+++ a/vendor/libssh/include/libssh/callbacks.h
@@ -25,7 +25,7 @@
 #ifndef _SSH_CALLBACK_H
 #define _SSH_CALLBACK_H
 
-#include "libssh.h"
+#include <libssh/libssh.h>
 #include <string.h>
 
 #ifdef __cplusplus
diff --git b/vendor/libssh/include/libssh/channels.h a/vendor/libssh/include/libssh/channels.h
index daafa1764..ce8540ae6 100644
--- b/vendor/libssh/include/libssh/channels.h
+++ a/vendor/libssh/include/libssh/channels.h
@@ -20,7 +20,7 @@
 
 #ifndef CHANNELS_H_
 #define CHANNELS_H_
-#include "priv.h"
+#include "libssh/priv.h"
 
 /**  @internal
  * Describes the different possible states in a
diff --git b/vendor/libssh/include/libssh/crypto.h a/vendor/libssh/include/libssh/crypto.h
index 50c4bbb7d..1d73613b4 100644
--- b/vendor/libssh/include/libssh/crypto.h
+++ a/vendor/libssh/include/libssh/crypto.h
@@ -33,7 +33,7 @@
 #elif defined(HAVE_LIBMBEDCRYPTO)
 #include <mbedtls/gcm.h>
 #endif
-#include "wrapper.h"
+#include "libssh/wrapper.h"
 
 #ifdef cbc_encrypt
 #undef cbc_encrypt
@@ -45,10 +45,10 @@
 #ifdef HAVE_OPENSSL_ECDH_H
 #include <openssl/ecdh.h>
 #endif
-#include "dh.h"
-#include "ecdh.h"
-#include "kex.h"
-#include "curve25519.h"
+#include "libssh/dh.h"
+#include "libssh/ecdh.h"
+#include "libssh/kex.h"
+#include "libssh/curve25519.h"
 
 #define DIGEST_MAX_LEN 64
 
diff --git b/vendor/libssh/include/libssh/dh.h a/vendor/libssh/include/libssh/dh.h
index 0011f2937..353dc2335 100644
--- b/vendor/libssh/include/libssh/dh.h
+++ a/vendor/libssh/include/libssh/dh.h
@@ -23,7 +23,7 @@
 
 #include "config.h"
 
-#include "crypto.h"
+#include "libssh/crypto.h"
 
 struct dh_ctx;
 
diff --git b/vendor/libssh/include/libssh/ecdh.h a/vendor/libssh/include/libssh/ecdh.h
index d594a3cec..17fe02e7b 100644
--- b/vendor/libssh/include/libssh/ecdh.h
+++ a/vendor/libssh/include/libssh/ecdh.h
@@ -22,7 +22,7 @@
 #define ECDH_H_
 
 #include "config.h"
-#include "callbacks.h"
+#include "libssh/callbacks.h"
 
 #ifdef HAVE_LIBCRYPTO
 #ifdef HAVE_OPENSSL_ECDH_H
diff --git b/vendor/libssh/include/libssh/ed25519.h a/vendor/libssh/include/libssh/ed25519.h
index b45f507d2..8a3263c8d 100644
--- b/vendor/libssh/include/libssh/ed25519.h
+++ a/vendor/libssh/include/libssh/ed25519.h
@@ -20,7 +20,7 @@
 
 #ifndef ED25519_H_
 #define ED25519_H_
-#include "priv.h"
+#include "libssh/priv.h"
 
 /**
  * @defgroup ed25519 ed25519 API
diff --git b/vendor/libssh/include/libssh/fe25519.h a/vendor/libssh/include/libssh/fe25519.h
index 87497b8d4..438d85db6 100644
--- b/vendor/libssh/include/libssh/fe25519.h
+++ a/vendor/libssh/include/libssh/fe25519.h
@@ -9,7 +9,7 @@
 #ifndef FE25519_H
 #define FE25519_H
 
-#include "priv.h"
+#include "libssh/priv.h"
 
 #define fe25519              crypto_sign_ed25519_ref_fe25519
 #define fe25519_freeze       crypto_sign_ed25519_ref_fe25519_freeze
diff --git b/vendor/libssh/include/libssh/kex.h a/vendor/libssh/include/libssh/kex.h
index 5e9946678..3a1f4a6f2 100644
--- b/vendor/libssh/include/libssh/kex.h
+++ a/vendor/libssh/include/libssh/kex.h
@@ -21,8 +21,8 @@
 #ifndef KEX_H_
 #define KEX_H_
 
-#include "priv.h"
-#include "callbacks.h"
+#include "libssh/priv.h"
+#include "libssh/callbacks.h"
 
 #define SSH_KEX_METHODS 10
 
diff --git b/vendor/libssh/include/libssh/keys.h a/vendor/libssh/include/libssh/keys.h
index 6b3681c6a..934189c27 100644
--- b/vendor/libssh/include/libssh/keys.h
+++ a/vendor/libssh/include/libssh/keys.h
@@ -22,8 +22,8 @@
 #define KEYS_H_
 
 #include "config.h"
-#include "libssh.h"
-#include "wrapper.h"
+#include "libssh/libssh.h"
+#include "libssh/wrapper.h"
 
 struct ssh_public_key_struct {
     int type;
diff --git b/vendor/libssh/include/libssh/libssh.h a/vendor/libssh/include/libssh/libssh.h
index 68c13cf01..7857a77bb 100644
--- b/vendor/libssh/include/libssh/libssh.h
+++ a/vendor/libssh/include/libssh/libssh.h
@@ -21,7 +21,7 @@
 #ifndef _LIBSSH_H
 #define _LIBSSH_H
 
-#include "libssh_version.h"
+#include <libssh/libssh_version.h>
 
 #if defined _WIN32 || defined __CYGWIN__
   #ifdef LIBSSH_STATIC
@@ -844,7 +844,7 @@ LIBSSH_API uint32_t ssh_buffer_get_len(ssh_buffer buffer);
 LIBSSH_API int ssh_session_set_disconnect_message(ssh_session session, const char *message);
 
 #ifndef LIBSSH_LEGACY_0_4
-#include "legacy.h"
+#include "libssh/legacy.h"
 #endif
 
 #ifdef __cplusplus
diff --git b/vendor/libssh/include/libssh/libsshpp.hpp a/vendor/libssh/include/libssh/libsshpp.hpp
index d12c49189..602c7aecb 100644
--- b/vendor/libssh/include/libssh/libsshpp.hpp
+++ a/vendor/libssh/include/libssh/libsshpp.hpp
@@ -51,8 +51,8 @@
 /* do not use deprecated functions */
 #define LIBSSH_LEGACY_0_4
 
-#include "libssh.h"
-#include "server.h"
+#include <libssh/libssh.h>
+#include <libssh/server.h>
 #include <stdlib.h>
 #include <stdarg.h>
 #include <stdio.h>
diff --git b/vendor/libssh/include/libssh/packet.h a/vendor/libssh/include/libssh/packet.h
index aab461e4b..dec3a439a 100644
--- b/vendor/libssh/include/libssh/packet.h
+++ a/vendor/libssh/include/libssh/packet.h
@@ -21,7 +21,7 @@
 #ifndef PACKET_H_
 #define PACKET_H_
 
-#include "wrapper.h"
+#include "libssh/wrapper.h"
 
 struct ssh_socket_struct;
 
diff --git b/vendor/libssh/include/libssh/pcap.h a/vendor/libssh/include/libssh/pcap.h
index 374825f35..2e43ae87a 100644
--- b/vendor/libssh/include/libssh/pcap.h
+++ a/vendor/libssh/include/libssh/pcap.h
@@ -22,7 +22,7 @@
 #define PCAP_H_
 
 #include "config.h"
-#include "libssh.h"
+#include "libssh/libssh.h"
 
 #ifdef WITH_PCAP
 typedef struct ssh_pcap_context_struct* ssh_pcap_context;
diff --git b/vendor/libssh/include/libssh/pki.h a/vendor/libssh/include/libssh/pki.h
index 0728f267f..1d5250fff 100644
--- b/vendor/libssh/include/libssh/pki.h
+++ a/vendor/libssh/include/libssh/pki.h
@@ -22,7 +22,7 @@
 #define PKI_H_
 
 #include <stdint.h>
-#include "priv.h"
+#include "libssh/priv.h"
 #ifdef HAVE_OPENSSL_EC_H
 #include <openssl/ec.h>
 #endif
@@ -32,13 +32,13 @@
 #ifdef HAVE_LIBCRYPTO
 #include <openssl/evp.h>
 #endif
-#include "crypto.h"
+#include "libssh/crypto.h"
 #if defined(HAVE_LIBCRYPTO) && defined(HAVE_OPENSSL_ED25519)
 /* If using OpenSSL implementation, define the signature lenght which would be
  * defined in libssh/ed25519.h otherwise */
 #define ED25519_SIG_LEN 64
 #else
-#include "ed25519.h"
+#include "libssh/ed25519.h"
 #endif
 /* This definition is used for both OpenSSL and internal implementations */
 #define ED25519_KEY_LEN 32
diff --git b/vendor/libssh/include/libssh/pki_priv.h a/vendor/libssh/include/libssh/pki_priv.h
index 13628c2d1..c0edb851c 100644
--- b/vendor/libssh/include/libssh/pki_priv.h
+++ a/vendor/libssh/include/libssh/pki_priv.h
@@ -21,7 +21,7 @@
 #ifndef PKI_PRIV_H_
 #define PKI_PRIV_H_
 
-#include "pki.h"
+#include "libssh/pki.h"
 
 /* defined in bcrypt_pbkdf.c */
 int bcrypt_pbkdf(const char *pass,
diff --git b/vendor/libssh/include/libssh/poly1305.h a/vendor/libssh/include/libssh/poly1305.h
index e1259502d..513f1b99a 100644
--- b/vendor/libssh/include/libssh/poly1305.h
+++ a/vendor/libssh/include/libssh/poly1305.h
@@ -5,7 +5,7 @@
 
 #ifndef POLY1305_H
 #define POLY1305_H
-#include "chacha20-poly1305-common.h"
+#include "libssh/chacha20-poly1305-common.h"
 
 void poly1305_auth(uint8_t out[POLY1305_TAGLEN], const uint8_t *m, size_t inlen,
     const uint8_t key[POLY1305_KEYLEN])
diff --git b/vendor/libssh/include/libssh/priv.h a/vendor/libssh/include/libssh/priv.h
index b21519034..bab761b0d 100644
--- b/vendor/libssh/include/libssh/priv.h
+++ a/vendor/libssh/include/libssh/priv.h
@@ -166,8 +166,8 @@ int ssh_gettimeofday(struct timeval *__p, void *__t);
 
 #endif /* _WIN32 */
 
-#include "libssh.h"
-#include "callbacks.h"
+#include "libssh/libssh.h"
+#include "libssh/callbacks.h"
 
 /* some constants */
 #ifndef PATH_MAX
diff --git b/vendor/libssh/include/libssh/server.h a/vendor/libssh/include/libssh/server.h
index 7c7b18ec4..c1a9c15cc 100644
--- b/vendor/libssh/include/libssh/server.h
+++ a/vendor/libssh/include/libssh/server.h
@@ -28,7 +28,7 @@
 #ifndef SERVER_H
 #define SERVER_H
 
-#include "libssh.h"
+#include "libssh/libssh.h"
 #define SERVERBANNER CLIENTBANNER
 
 #ifdef __cplusplus
diff --git b/vendor/libssh/include/libssh/session.h a/vendor/libssh/include/libssh/session.h
index 124687257..d3e5787cc 100644
--- b/vendor/libssh/include/libssh/session.h
+++ a/vendor/libssh/include/libssh/session.h
@@ -22,16 +22,16 @@
 #define SESSION_H_
 #include <stdbool.h>
 
-#include "priv.h"
-#include "callbacks.h"
-#include "kex.h"
-#include "packet.h"
-#include "pcap.h"
-#include "auth.h"
-#include "channels.h"
-#include "poll.h"
-#include "config.h"
-#include "misc.h"
+#include "libssh/priv.h"
+#include "libssh/callbacks.h"
+#include "libssh/kex.h"
+#include "libssh/packet.h"
+#include "libssh/pcap.h"
+#include "libssh/auth.h"
+#include "libssh/channels.h"
+#include "libssh/poll.h"
+#include "libssh/config.h"
+#include "libssh/misc.h"
 
 /* These are the different states a SSH session can be into its life */
 enum ssh_session_state_e {
diff --git b/vendor/libssh/include/libssh/socket.h a/vendor/libssh/include/libssh/socket.h
index d9766594a..ae56f5f6e 100644
--- b/vendor/libssh/include/libssh/socket.h
+++ a/vendor/libssh/include/libssh/socket.h
@@ -21,7 +21,7 @@
 #ifndef SOCKET_H_
 #define SOCKET_H_
 
-#include "callbacks.h"
+#include "libssh/callbacks.h"
 struct ssh_poll_handle_struct;
 /* socket.c */
 
diff --git b/vendor/libssh/include/libssh/string.h a/vendor/libssh/include/libssh/string.h
index e74b80876..8c7db1df0 100644
--- b/vendor/libssh/include/libssh/string.h
+++ a/vendor/libssh/include/libssh/string.h
@@ -20,7 +20,7 @@
 
 #ifndef STRING_H_
 #define STRING_H_
-#include "priv.h"
+#include "libssh/priv.h"
 
 /* must be 32 bits number + immediately our data */
 #ifdef _MSC_VER
diff --git b/vendor/libssh/include/libssh/threads.h a/vendor/libssh/include/libssh/threads.h
index 7ba45706c..522f91d5b 100644
--- b/vendor/libssh/include/libssh/threads.h
+++ a/vendor/libssh/include/libssh/threads.h
@@ -21,8 +21,8 @@
 #ifndef THREADS_H_
 #define THREADS_H_
 
-#include "libssh.h"
-#include "callbacks.h"
+#include <libssh/libssh.h>
+#include <libssh/callbacks.h>
 
 #if HAVE_PTHREAD
 
diff --git b/vendor/libssh/include/libssh/wrapper.h a/vendor/libssh/include/libssh/wrapper.h
index 2275032eb..f4a33d2dc 100644
--- b/vendor/libssh/include/libssh/wrapper.h
+++ a/vendor/libssh/include/libssh/wrapper.h
@@ -24,10 +24,10 @@
 #include <stdbool.h>
 
 #include "config.h"
-#include "libssh.h"
-#include "libcrypto.h"
-#include "libgcrypt.h"
-#include "libmbedcrypto.h"
+#include "libssh/libssh.h"
+#include "libssh/libcrypto.h"
+#include "libssh/libgcrypt.h"
+#include "libssh/libmbedcrypto.h"
 
 enum ssh_kdf_digest {
     SSH_KDF_SHA1=1,
