commit cb860b78c291b1bd3cfc38f6f2a6af3db81ec809
Author: Niels Martignène <niels.martignene@protonmail.com>
Date:   Thu Feb 12 10:44:46 2026 +0100

    10

diff --git a/vendor/libssh/src/buffer.c b/vendor/libssh/src/buffer.c
index 601191755f..caab6334de 100644
--- a/vendor/libssh/src/buffer.c
+++ b/vendor/libssh/src/buffer.c
@@ -28,7 +28,9 @@
 #include <stdbool.h>
 #include <stdio.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/legacy.c b/vendor/libssh/src/legacy.c
index 5fad635f9f..a2a39a6d21 100644
--- a/vendor/libssh/src/legacy.c
+++ b/vendor/libssh/src/legacy.c
@@ -26,9 +26,16 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <errno.h>
 #include <stdio.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
+
 #include <libssh/priv.h>
 #include <libssh/session.h>
 #include <libssh/server.h>
diff --git a/vendor/libssh/src/messages.c b/vendor/libssh/src/messages.c
index 824b2772a4..47cd402956 100644
--- a/vendor/libssh/src/messages.c
+++ b/vendor/libssh/src/messages.c
@@ -26,7 +26,9 @@
 #include <string.h>
 #include <stdlib.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/packet_cb.c b/vendor/libssh/src/packet_cb.c
index 6228b44a48..162b9dbb67 100644
--- a/vendor/libssh/src/packet_cb.c
+++ b/vendor/libssh/src/packet_cb.c
@@ -24,6 +24,9 @@
 #include "config.h"
 
 #include <stdlib.h>
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #ifdef HAVE_ARPA_INET_H
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/packet_crypt.c b/vendor/libssh/src/packet_crypt.c
index 12846fe583..02a3454b4c 100644
--- a/vendor/libssh/src/packet_crypt.c
+++ b/vendor/libssh/src/packet_crypt.c
@@ -27,7 +27,9 @@
 #include <stdio.h>
 #include <string.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #endif
diff --git a/vendor/libssh/src/pki.c b/vendor/libssh/src/pki.c
index 76fde83837..f8d02ddfcd 100644
--- a/vendor/libssh/src/pki.c
+++ b/vendor/libssh/src/pki.c
@@ -35,6 +35,9 @@
 #include "config.h"
 #include "libssh/wrapper.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <errno.h>
 #include <ctype.h>
 #include <stdint.h>
diff --git a/vendor/libssh/src/sftp_aio.c b/vendor/libssh/src/sftp_aio.c
index 20e9853af8..0dc5467d7c 100644
--- a/vendor/libssh/src/sftp_aio.c
+++ b/vendor/libssh/src/sftp_aio.c
@@ -25,6 +25,10 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
+
 #include "libssh/sftp.h"
 #include "libssh/sftp_priv.h"
 #include "libssh/buffer.h"
diff --git a/vendor/libssh/src/sftp_common.c b/vendor/libssh/src/sftp_common.c
index 02970e4a47..3b400e771a 100644
--- a/vendor/libssh/src/sftp_common.c
+++ b/vendor/libssh/src/sftp_common.c
@@ -26,6 +26,9 @@
 
 #include "config.h"
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #include <ctype.h>
 
 #include "libssh/sftp.h"
diff --git a/vendor/libssh/src/string.c b/vendor/libssh/src/string.c
index e78737b3eb..5d7f7f8452 100644
--- a/vendor/libssh/src/string.c
+++ b/vendor/libssh/src/string.c
@@ -26,7 +26,9 @@
 #include <errno.h>
 #include <limits.h>
 
-#ifndef _WIN32
+#ifdef _WIN32
+#include <winsock2.h>
+#else
 #include <arpa/inet.h>
 #include <netinet/in.h>
 #endif
diff --git a/vendor/libssh/src/ttyopts.c b/vendor/libssh/src/ttyopts.c
index 251a988bb7..0d01fba1ac 100644
--- a/vendor/libssh/src/ttyopts.c
+++ b/vendor/libssh/src/ttyopts.c
@@ -27,9 +27,16 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
+
 #include <libssh/priv.h>
 #include <string.h>
 
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 #ifdef HAVE_TERMIOS_H
 #include <termios.h>
 #endif
