diff --git a/vendor/libsodium/src/libsodium/include/sodium/randombytes.h b/vendor/libsodium/src/libsodium/include/sodium/randombytes.h
index c83a4df50b..63a7edc1b1 100644
--- a/vendor/libsodium/src/libsodium/include/sodium/randombytes.h
+++ b/vendor/libsodium/src/libsodium/include/sodium/randombytes.h
@@ -59,12 +59,6 @@ int randombytes_set_implementation(const randombytes_implementation *impl)
 SODIUM_EXPORT
 const char *randombytes_implementation_name(void);
 
-/* -- NaCl compatibility interface -- */
-
-SODIUM_EXPORT
-void randombytes(unsigned char * const buf, const unsigned long long buf_len)
-            __attribute__ ((nonnull));
-
 #ifdef __cplusplus
 }
 #endif
diff --git a/vendor/libsodium/src/libsodium/randombytes/randombytes.c b/vendor/libsodium/src/libsodium/randombytes/randombytes.c
index 85c244549d..22c1139990 100644
--- a/vendor/libsodium/src/libsodium/randombytes/randombytes.c
+++ b/vendor/libsodium/src/libsodium/randombytes/randombytes.c
@@ -207,10 +207,3 @@ randombytes_close(void)
     }
     return 0;
 }
-
-void
-randombytes(unsigned char * const buf, const unsigned long long buf_len)
-{
-    assert(buf_len <= SIZE_MAX);
-    randombytes_buf(buf, (size_t) buf_len);
-}
