diff --git a/vendor/libseccomp/include/seccomp.h b/vendor/libseccomp/include/seccomp.h
index 5677430f5..b95803ab7 100644
--- a/vendor/libseccomp/include/seccomp.h
+++ b/vendor/libseccomp/include/seccomp.h
@@ -818,7 +818,7 @@ int seccomp_export_bpf(const scmp_filter_ctx ctx, int fd);
 #define __NR_SCMP_ERROR		-1
 #define __NR_SCMP_UNDEF		-2
 
-#include <seccomp-syscalls.h>
+#include "seccomp-syscalls.h"
 
 #ifdef __cplusplus
 }
