diff --git a/vendor/libsodium/src/libsodium/randombytes/internal/randombytes_internal_random.c b/vendor/libsodium/src/libsodium/randombytes/internal/randombytes_internal_random.c
index b3c9a75d..649c4b20 100644
--- a/vendor/libsodium/src/libsodium/randombytes/internal/randombytes_internal_random.c
+++ b/vendor/libsodium/src/libsodium/randombytes/internal/randombytes_internal_random.c
@@ -96,7 +96,7 @@ BOOLEAN NTAPI RtlGenRandom(PVOID RandomBuffer, ULONG RandomBufferLength);
 #endif
 
 #ifndef TLS
-# ifdef _WIN32
+# ifdef _MSC_VER
 #  define TLS __declspec(thread)
 # else
 #  define TLS
