diff --git a/vendor/libssh/src/misc.c b/vendor/libssh/src/misc.c
index 6ae7cdf13..54a411bc4 100644
--- a/vendor/libssh/src/misc.c
+++ b/vendor/libssh/src/misc.c
@@ -185,14 +185,14 @@ char *ssh_get_local_username(void)
     int rc;
 
     /* get the size */
-    GetUserName(NULL, &size);
+    GetUserNameA(NULL, &size);
 
     user = (char *)malloc(size);
     if (user == NULL) {
         return NULL;
     }
 
-    if (GetUserName(user, &size)) {
+    if (GetUserNameA(user, &size)) {
         rc = ssh_check_username_syntax(user);
         if (rc == SSH_OK) {
             return user;
