diff --git a/vendor/lit-html/lit/packages/lit-html/src/lit-html.ts b/vendor/lit-html/lit/packages/lit-html/src/lit-html.ts
index bf5efc13..6057080e 100644
--- a/vendor/lit-html/lit/packages/lit-html/src/lit-html.ts
+++ b/vendor/lit-html/lit/packages/lit-html/src/lit-html.ts
@@ -260,11 +260,10 @@ const trustedTypes = (global as unknown as TrustedTypesWindow).trustedTypes;
  * before any untrusted expressions have been mixed in. Therefor it is
  * considered safe by construction.
  */
-const policy = trustedTypes
-  ? trustedTypes.createPolicy('lit-html', {
-      createHTML: (s) => s,
-    })
-  : undefined;
+const policy =
+  trustedTypes?.createPolicy?.('lit-html', {
+    createHTML: (s) => s,
+  }) ?? undefined;
