diff --git a/vendor/miniz/miniz.c b/vendor/miniz/miniz.c
index c197c181c..a5eedcc5d 100644
--- a/vendor/miniz/miniz.c
+++ b/vendor/miniz/miniz.c
@@ -3182,7 +3182,9 @@ static int mz_stat64(const char *path, struct __stat64 *buffer)
 #define MZ_DELETE_FILE remove
 
 #else
+#if _FILE_OFFSET_BITS != 64
 #pragma message("Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.")
+#endif
 #ifndef MINIZ_NO_TIME
 #include <utime.h>
 #endif
