diff --git a/vendor/miniz/miniz.c b/vendor/miniz/miniz.c
index 87bdedb18..5d4ffd7ce 100644
--- a/vendor/miniz/miniz.c
+++ b/vendor/miniz/miniz.c
@@ -3086,7 +3086,9 @@ static FILE *mz_freopen(const char *pPath, const char *pMode, FILE *pStream)
 #define MZ_DELETE_FILE remove
 
 #else
-#pragma message("Using fopen, ftello, fseeko, stat() etc. path for file I/O - this path may not support large files.")
+#if _FILE_OFFSET_BITS != 64
+#pragma message("Using fopen, ftello, fseeko, stat() etc. path for file I/O without -D_FILE_OFFSET_BITS=64 - this path may not support large files.")
+#endif
 #ifndef MINIZ_NO_TIME
 #include <utime.h>
 #endif
