diff --git a/vendor/quickjs/quickjs.c b/vendor/quickjs/quickjs.c
index 1ac3052b8..7d265f607 100644
--- a/vendor/quickjs/quickjs.c
+++ b/vendor/quickjs/quickjs.c
@@ -72,10 +72,10 @@
 #define CONFIG_ATOMICS
 #endif
 
-#if !defined(EMSCRIPTEN) && !defined(_MSC_VER)
-/* enable stack limitation */
-#define CONFIG_STACK_CHECK
-#endif
+// #if !defined(EMSCRIPTEN) && !defined(_MSC_VER)
+// /* enable stack limitation */
+// #define CONFIG_STACK_CHECK
+// #endif
 
 
 /* dump object free */
