diff --git b/vendor/re2/re2/bitmap256.h a/vendor/re2/re2/bitmap256.h
index 4899379e..60f03b2e 100644
--- b/vendor/re2/re2/bitmap256.h
+++ a/vendor/re2/re2/bitmap256.h
@@ -11,8 +11,8 @@
 #include <stdint.h>
 #include <string.h>
 
-#include "util/util.h"
-#include "util/logging.h"
+#include "../util/util.h"
+#include "../util/logging.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/filtered_re2.h a/vendor/re2/re2/filtered_re2.h
index 86fa5868..610d8b28 100644
--- b/vendor/re2/re2/filtered_re2.h
+++ a/vendor/re2/re2/filtered_re2.h
@@ -24,7 +24,7 @@
 #include <string>
 #include <vector>
 
-#include "re2/re2.h"
+#include "re2.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/prefilter.h a/vendor/re2/re2/prefilter.h
index 4fedeb4a..44d5ee35 100644
--- b/vendor/re2/re2/prefilter.h
+++ a/vendor/re2/re2/prefilter.h
@@ -13,8 +13,8 @@
 #include <string>
 #include <vector>
 
-#include "util/util.h"
-#include "util/logging.h"
+#include "../util/util.h"
+#include "../util/logging.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/prefilter_tree.h a/vendor/re2/re2/prefilter_tree.h
index 5d73074d..b568dda8 100644
--- b/vendor/re2/re2/prefilter_tree.h
+++ a/vendor/re2/re2/prefilter_tree.h
@@ -20,9 +20,9 @@
 #include <string>
 #include <vector>
 
-#include "util/util.h"
-#include "re2/prefilter.h"
-#include "re2/sparse_array.h"
+#include "../util/util.h"
+#include "prefilter.h"
+#include "sparse_array.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/prog.h a/vendor/re2/re2/prog.h
index 4306672e..d0bdfce2 100644
--- b/vendor/re2/re2/prog.h
+++ a/vendor/re2/re2/prog.h
@@ -16,12 +16,12 @@
 #include <vector>
 #include <type_traits>
 
-#include "util/util.h"
-#include "util/logging.h"
-#include "re2/pod_array.h"
-#include "re2/re2.h"
-#include "re2/sparse_array.h"
-#include "re2/sparse_set.h"
+#include "../util/util.h"
+#include "../util/logging.h"
+#include "pod_array.h"
+#include "re2.h"
+#include "sparse_array.h"
+#include "sparse_set.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/re2.h a/vendor/re2/re2/re2.h
index 32b2718f..d6e2ce89 100644
--- b/vendor/re2/re2/re2.h
+++ a/vendor/re2/re2/re2.h
@@ -211,7 +211,7 @@
 #include <TargetConditionals.h>
 #endif
 
-#include "re2/stringpiece.h"
+#include "stringpiece.h"
 
 namespace re2 {
 class Prog;
diff --git b/vendor/re2/re2/regexp.h a/vendor/re2/re2/regexp.h
index a5d85c81..9b7f4645 100644
--- b/vendor/re2/re2/regexp.h
+++ a/vendor/re2/re2/regexp.h
@@ -91,10 +91,10 @@
 #include <set>
 #include <string>
 
-#include "util/util.h"
-#include "util/logging.h"
-#include "util/utf.h"
-#include "re2/stringpiece.h"
+#include "../util/util.h"
+#include "../util/logging.h"
+#include "../util/utf.h"
+#include "stringpiece.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/set.h a/vendor/re2/re2/set.h
index 59733fd9..54d666be 100644
--- b/vendor/re2/re2/set.h
+++ a/vendor/re2/re2/set.h
@@ -9,7 +9,7 @@
 #include <utility>
 #include <vector>
 
-#include "re2/re2.h"
+#include "re2.h"
 
 namespace re2 {
 class Prog;
diff --git b/vendor/re2/re2/sparse_array.h a/vendor/re2/re2/sparse_array.h
index 09ffe086..2ceab1b3 100644
--- b/vendor/re2/re2/sparse_array.h
+++ a/vendor/re2/re2/sparse_array.h
@@ -102,7 +102,7 @@
 #include <memory>
 #include <utility>
 
-#include "re2/pod_array.h"
+#include "pod_array.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/sparse_set.h a/vendor/re2/re2/sparse_set.h
index 06ed88d8..247877e3 100644
--- b/vendor/re2/re2/sparse_set.h
+++ a/vendor/re2/re2/sparse_set.h
@@ -61,7 +61,7 @@
 #include <memory>
 #include <utility>
 
-#include "re2/pod_array.h"
+#include "pod_array.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/unicode_casefold.h a/vendor/re2/re2/unicode_casefold.h
index 8bdbb42f..3d569cd7 100644
--- b/vendor/re2/re2/unicode_casefold.h
+++ a/vendor/re2/re2/unicode_casefold.h
@@ -41,8 +41,8 @@
 
 #include <stdint.h>
 
-#include "util/util.h"
-#include "util/utf.h"
+#include "../util/util.h"
+#include "../util/utf.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/unicode_groups.h a/vendor/re2/re2/unicode_groups.h
index 75f55daa..8617c0b5 100644
--- b/vendor/re2/re2/unicode_groups.h
+++ a/vendor/re2/re2/unicode_groups.h
@@ -20,8 +20,8 @@
 
 #include <stdint.h>
 
-#include "util/util.h"
-#include "util/utf.h"
+#include "../util/util.h"
+#include "../util/utf.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/re2/walker-inl.h a/vendor/re2/re2/walker-inl.h
index 310be54b..f5a38655 100644
--- b/vendor/re2/re2/walker-inl.h
+++ a/vendor/re2/re2/walker-inl.h
@@ -15,8 +15,8 @@
 
 #include <stack>
 
-#include "util/logging.h"
-#include "re2/regexp.h"
+#include "../util/logging.h"
+#include "regexp.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/util/benchmark.h a/vendor/re2/util/benchmark.h
index d97b49e1..23397380 100644
--- b/vendor/re2/util/benchmark.h
+++ a/vendor/re2/util/benchmark.h
@@ -8,8 +8,8 @@
 #include <stdint.h>
 #include <functional>
 
-#include "util/logging.h"
-#include "util/util.h"
+#include "logging.h"
+#include "util.h"
 
 // Globals for the old benchmark API.
 void StartBenchmarkTiming();
diff --git b/vendor/re2/util/logging.h a/vendor/re2/util/logging.h
index 5b2217f2..c9d8dbc6 100644
--- b/vendor/re2/util/logging.h
+++ a/vendor/re2/util/logging.h
@@ -13,7 +13,7 @@
 #include <ostream>
 #include <sstream>
 
-#include "util/util.h"
+#include "util.h"
 
 // Debug-only checking.
 #define DCHECK(condition) assert(condition)
diff --git b/vendor/re2/util/pcre.h a/vendor/re2/util/pcre.h
index 644dce68..85b0ee35 100644
--- b/vendor/re2/util/pcre.h
+++ a/vendor/re2/util/pcre.h
@@ -161,8 +161,8 @@
 //         Octal(&a), Hex(&b), CRadix(&c), CRadix(&d));
 // will leave 64 in a, b, c, and d.
 
-#include "util/util.h"
-#include "re2/stringpiece.h"
+#include "util.h"
+#include "../re2/stringpiece.h"
 
 #ifdef USEPCRE
 #include <pcre.h>
diff --git b/vendor/re2/util/strutil.h a/vendor/re2/util/strutil.h
index a69908a0..5c55174e 100644
--- b/vendor/re2/util/strutil.h
+++ a/vendor/re2/util/strutil.h
@@ -7,8 +7,8 @@
 
 #include <string>
 
-#include "re2/stringpiece.h"
-#include "util/util.h"
+#include "../re2/stringpiece.h"
+#include "util.h"
 
 namespace re2 {
 
diff --git b/vendor/re2/util/test.h a/vendor/re2/util/test.h
index 54e6f8fb..bac5176f 100644
--- b/vendor/re2/util/test.h
+++ a/vendor/re2/util/test.h
@@ -5,8 +5,8 @@
 #ifndef UTIL_TEST_H_
 #define UTIL_TEST_H_
 
-#include "util/util.h"
-#include "util/logging.h"
+#include "util.h"
+#include "logging.h"
 
 namespace testing {
 std::string TempDir();
