diff --git a/vendor/sqlite3/sqlite3mc.c b/vendor/sqlite3/sqlite3mc.c
index 50347641e..009717623 100644
--- a/vendor/sqlite3/sqlite3mc.c
+++ b/vendor/sqlite3/sqlite3mc.c
@@ -273440,12 +273440,8 @@ void RijndaelDecrypt(Rijndael* rijndael, UINT8 a[16], UINT8 b[16]);
 
 #ifndef SQLITE3MC_OMIT_AES_HARDWARE_SUPPORT
 
-#if defined __ARM_FEATURE_CRYPTO
-#define HAS_AES_HARDWARE AES_HARDWARE_NEON
-
-
 /* --- CLang --- */
-#elif defined(__clang__)
+#if defined(__clang__)
 
 #if __has_attribute(target) && __has_include(<wmmintrin.h>) && (defined(__x86_64__) || defined(__i386))
 #define HAS_AES_HARDWARE AES_HARDWARE_NI
@@ -273536,6 +273532,9 @@ toUint32FromLE(const void* buffer)
 
 #include <cpuid.h>
 
+/* FUNCtion attributes for ISA (Instruction Set Architecture) */
+#define FUNC_ISA __attribute__ ((target("aes,sse4.1")))
+
 static int
 aesHardwareCheck()
 {
@@ -273550,6 +273549,9 @@ aesHardwareCheck()
 
 #include <intrin.h>
 
+/* FUNCtion attributes for ISA (Instruction Set Architecture) */
+#define FUNC_ISA
+
 static int
 aesHardwareCheck()
 {
@@ -273563,7 +273565,7 @@ aesHardwareCheck()
 #include <wmmintrin.h>
 #include <smmintrin.h>
 
-static int
+static FUNC_ISA int
 aesGenKeyEncryptInternal(const unsigned char* userKey, const int bits, __m128i* keyData)
 {
   int rc = 0;
@@ -273616,7 +273618,7 @@ aesGenKeyEncryptInternal(const unsigned char* userKey, const int bits, __m128i*
   return rc;
 }
 
-static int
+static FUNC_ISA int
 aesGenKeyEncrypt(const unsigned char* userKey, const int bits, unsigned char* keyData)
 {
   int numberOfRounds = (bits == 128) ? 10 : (bits == 192) ? 12 : (bits == 256) ? 14 : 0;
@@ -273638,7 +273640,7 @@ aesGenKeyEncrypt(const unsigned char* userKey, const int bits, unsigned char* ke
   return rc;
 }
 
-static int
+static FUNC_ISA int
 aesGenKeyDecrypt(const unsigned char* userKey, const int bits, unsigned char* keyData)
 {
   int numberOfRounds = (bits == 128) ? 10 : (bits == 192) ? 12 : (bits == 256) ? 14 : 0;
@@ -273672,7 +273674,7 @@ aesGenKeyDecrypt(const unsigned char* userKey, const int bits, unsigned char* ke
 ** AES CBC CTS Encryption
 */
 
-static void
+static FUNC_ISA void
 aesEncryptCBC(const unsigned char* in,
               unsigned char* out,
               unsigned char ivec[16],
@@ -273739,7 +273741,7 @@ aesEncryptCBC(const unsigned char* in,
 /*
 ** AES CBC CTS decryption
 */
-static void
+static FUNC_ISA void
 aesDecryptCBC(const unsigned char* in,
               unsigned char* out,
               unsigned char ivec[16],
@@ -273829,18 +273831,7 @@ aesDecryptCBC(const unsigned char* in,
 #elif HAS_AES_HARDWARE == AES_HARDWARE_NEON
 /* --- Implementation for AES-NEON --- */
 
-/* Set target architecture manually, if necessary */
-#ifdef USE_CLANG_ATTR_TARGET_AARCH64
-#define __ARM_NEON 1
-#define __ARM_FEATURE_CRYPTO 1
-#define __ARM_FEATURE_AES 1
 #define FUNC_ISA __attribute__ ((target("neon,crypto")))
-#endif /* USE_CLANG_ATTR_TARGET_AARCH64 */
-
-/* FUNCtion attributes for ISA (Instruction Set Architecture) */
-#ifndef FUNC_ISA
-#define FUNC_ISA
-#endif
 
 #ifdef USE_ARM64_NEON_H
 #include <arm64_neon.h>
