diff --git a/vendor/sqlite3mc/wasm/jswasm/sqlite3-worker1-bundler-friendly.mjs b/vendor/sqlite3mc/wasm/jswasm/sqlite3-worker1-bundler-friendly.mjs
index 7eaf78a7eb..0db4ef2618 100644
--- a/vendor/sqlite3mc/wasm/jswasm/sqlite3-worker1-bundler-friendly.mjs
+++ b/vendor/sqlite3mc/wasm/jswasm/sqlite3-worker1-bundler-friendly.mjs
@@ -31,5 +31,5 @@
   - `sqlite3.dir`, if set, treats the given directory name as the
     directory from which `sqlite3.js` will be loaded.
 */
-import {default as sqlite3InitModule} from './sqlite3-bundler-friendly.mjs';
+import {default as sqlite3InitModule} from './sqlite3.js';
 sqlite3InitModule().then(sqlite3 => sqlite3.initWorker1API());
