diff --git a/vendor/sqlite3mc/sqlite3mc.c b/vendor/sqlite3mc/sqlite3mc.c
index 817c88e95..76cfa827e 100644
--- a/vendor/sqlite3mc/sqlite3mc.c
+++ b/vendor/sqlite3mc/sqlite3mc.c
@@ -284995,7 +284995,7 @@ softaes_block_encrypt(const SoftAesBlock block, const SoftAesBlock rk)
 }
 #else
 
-uint32_t _aes_lut[256] __attribute__((visibility("hidden"))) = {
+static uint32_t _aes_lut[256] __attribute__((visibility("hidden"))) = {
     0xa56363c6, 0x847c7cf8, 0x997777ee, 0x8d7b7bf6, 0x0df2f2ff, 0xbd6b6bd6, 0xb16f6fde, 0x54c5c591,
     0x50303060, 0x03010102, 0xa96767ce, 0x7d2b2b56, 0x19fefee7, 0x62d7d7b5, 0xe6abab4d, 0x9a7676ec,
     0x45caca8f, 0x9d82821f, 0x40c9c989, 0x877d7dfa, 0x15fafaef, 0xeb5959b2, 0xc947478e, 0x0bf0f0fb,
@@ -285224,7 +285224,7 @@ typedef struct aegis128l_implementation {
 /* #include "implementations.h" */
 
 
-extern struct aegis128l_implementation aegis128l_aesni_implementation;
+// extern struct aegis128l_implementation aegis128l_aesni_implementation;
 
 #endif /* AEGIS128L_AESNI_H */
 /*** End of #include "aegis128l_aesni.h" ***/
@@ -286077,7 +286077,7 @@ AEGIS_state_mac_clone(aegis128l_mac_state *dst, const aegis128l_mac_state *src)
 /*** End of #include "aegis128l_common.h" ***/
 
 
-struct aegis128l_implementation aegis128l_aesni_implementation = {
+static struct aegis128l_implementation aegis128l_aesni_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -287215,7 +287215,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis128l_implementation aegis128l_armcrypto_implementation;
+// extern struct aegis128l_implementation aegis128l_armcrypto_implementation;
 
 #endif /* AEGIS128L_ARMCRYPTO_H */
 /*** End of #include "aegis128l_armcrypto.h" ***/
@@ -288076,7 +288076,7 @@ AEGIS_state_mac_clone(aegis128l_mac_state *dst, const aegis128l_mac_state *src)
 /*** End of #include "aegis128l_common.h" ***/
 
 
-struct aegis128l_implementation aegis128l_armcrypto_implementation = {
+static struct aegis128l_implementation aegis128l_armcrypto_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -288222,7 +288222,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis128l_implementation aegis128l_soft_implementation;
+// extern struct aegis128l_implementation aegis128l_soft_implementation;
 
 #endif /* AEGIS128L_SOFT_H */
 /*** End of #include "aegis128l_soft.h" ***/
@@ -289066,7 +289066,7 @@ AEGIS_state_mac_clone(aegis128l_mac_state *dst, const aegis128l_mac_state *src)
 /*** End of #include "aegis128l_common.h" ***/
 
 
-struct aegis128l_implementation aegis128l_soft_implementation = {
+static struct aegis128l_implementation aegis128l_soft_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -294017,7 +294017,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis128x2_implementation aegis128x2_soft_implementation;
+// extern struct aegis128x2_implementation aegis128x2_soft_implementation;
 
 #endif /* AEGIS128X2_SOFT_H */
 /*** End of #include "aegis128x2_soft.h" ***/
@@ -294973,7 +294973,7 @@ AEGIS_state_mac_clone(aegis128x2_mac_state *dst, const aegis128x2_mac_state *src
 /*** End of #include "aegis128x2_common.h" ***/
 
 
-struct aegis128x2_implementation aegis128x2_soft_implementation = {
+static struct aegis128x2_implementation aegis128x2_soft_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -301145,7 +301145,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis128x4_implementation aegis128x4_soft_implementation;
+// extern struct aegis128x4_implementation aegis128x4_soft_implementation;
 
 #endif /* AEGIS128X4_SOFT_H */
 /*** End of #include "aegis128x4_soft.h" ***/
@@ -302123,7 +302123,7 @@ AEGIS_state_mac_clone(aegis128x4_mac_state *dst, const aegis128x4_mac_state *src
 /*** End of #include "aegis128x4_common.h" ***/
 
 
-struct aegis128x4_implementation aegis128x4_soft_implementation = {
+static struct aegis128x4_implementation aegis128x4_soft_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -302639,7 +302639,7 @@ typedef struct aegis256_implementation {
 /* #include "implementations.h" */
 
 
-extern struct aegis256_implementation aegis256_aesni_implementation;
+// extern struct aegis256_implementation aegis256_aesni_implementation;
 
 #endif /* AEGIS256_AESNI_H */
 /*** End of #include "aegis256_aesni.h" ***/
@@ -303474,7 +303474,7 @@ AEGIS_state_mac_clone(aegis256_mac_state *dst, const aegis256_mac_state *src)
 /*** End of #include "aegis256_common.h" ***/
 
 
-struct aegis256_implementation aegis256_aesni_implementation = {
+static struct aegis256_implementation aegis256_aesni_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -304596,7 +304596,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis256_implementation aegis256_armcrypto_implementation;
+// extern struct aegis256_implementation aegis256_armcrypto_implementation;
 
 #endif /* AEGIS256_ARMCRYPTO_H */
 /*** End of #include "aegis256_armcrypto.h" ***/
@@ -305441,7 +305441,7 @@ AEGIS_state_mac_clone(aegis256_mac_state *dst, const aegis256_mac_state *src)
 /*** End of #include "aegis256_common.h" ***/
 
 
-struct aegis256_implementation aegis256_armcrypto_implementation = {
+static struct aegis256_implementation aegis256_armcrypto_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
@@ -305587,7 +305587,7 @@ AEGIS_API_IMPL_LIST_MAC
 /* #include "implementations.h" */
 
 
-extern struct aegis256_implementation aegis256_soft_implementation;
+// extern struct aegis256_implementation aegis256_soft_implementation;
 
 #endif /* AEGIS256_SOFT_H */
 /*** End of #include "aegis256_soft.h" ***/
@@ -306412,7 +306412,7 @@ AEGIS_state_mac_clone(aegis256_mac_state *dst, const aegis256_mac_state *src)
 /*** End of #include "aegis256_common.h" ***/
 
 
-struct aegis256_implementation aegis256_soft_implementation = {
+static struct aegis256_implementation aegis256_soft_implementation = {
 /* #include "../common/func_table.h" */
 /*** Begin of #include "../common/func_table.h" ***/
 /*
