diff --git a/vendor/tkspline/src/spline.h b/vendor/tkspline/src/spline.h
index e7bdf150a..dfdccd24f 100644
--- a/vendor/tkspline/src/spline.h
+++ b/vendor/tkspline/src/spline.h
@@ -105,13 +105,13 @@ public:
     }
 
     // optional, but if called it has to come be before set_points()
-    void set_boundary(bd_type left, double left_value,
-                      bd_type right, double right_value,
-                      bool force_linear_extrapolation=false);
-    void set_points(const std::vector<double>& x,
-                    const std::vector<double>& y, bool cubic_spline=true);
-    double operator() (double x) const;
-    double deriv(int order, double x) const;
+    [[maybe_unused]] void set_boundary(bd_type left, double left_value,
+                                       bd_type right, double right_value,
+                                       bool force_linear_extrapolation=false);
+    [[maybe_unused]] void set_points(const std::vector<double>& x,
+                                     const std::vector<double>& y, bool cubic_spline=true);
+    [[maybe_unused]] double operator() (double x) const;
+    [[maybe_unused]] double deriv(int order, double x) const;
 };
 
 
