diff --git a/vendor/webkit/webkit/Source/WTF/wtf/CMakeLists.txt b/vendor/webkit/webkit/Source/WTF/wtf/CMakeLists.txt
--- a/vendor/webkit/webkit/Source/WTF/wtf/CMakeLists.txt
+++ b/vendor/webkit/webkit/Source/WTF/wtf/CMakeLists.txt
@@ -736,9 +736,9 @@ set(WTF_PRIVATE_INCLUDE_DIRECTORIES
 
 set(WTF_LIBRARIES
     ${CMAKE_DL_LIBS}
-    ICU::data
     ICU::i18n
     ICU::uc
+    ICU::data
 )
 
 if (NOT USE_SYSTEM_MALLOC)
