diff --git a/vendor/wren/src/include/wren.h b/vendor/wren/src/include/wren.h
index 16abd9b31..6c6bec9b1 100644
--- a/vendor/wren/src/include/wren.h
+++ b/vendor/wren/src/include/wren.h
@@ -548,6 +548,9 @@ WREN_API bool wrenHasModule(WrenVM* vm, const char* module);
 // runtime error object.
 WREN_API void wrenAbortFiber(WrenVM* vm, int slot);
 
+// Returns true if the current fiber is about to abort.
+WREN_API bool wrenWillAbort(WrenVM *vm);
+
 // Returns the user data associated with the WrenVM.
 WREN_API void* wrenGetUserData(WrenVM* vm);
 
diff --git a/vendor/wren/src/vm/wren_vm.c b/vendor/wren/src/vm/wren_vm.c
index c2c8e6a06..3d7dcc96b 100644
--- a/vendor/wren/src/vm/wren_vm.c
+++ b/vendor/wren/src/vm/wren_vm.c
@@ -1996,6 +1996,11 @@ void wrenAbortFiber(WrenVM* vm, int slot)
   vm->fiber->error = vm->apiStack[slot];
 }
 
+bool wrenWillAbort(WrenVM* vm)
+{
+  return wrenHasError(vm->fiber);
+}
+
 void* wrenGetUserData(WrenVM* vm)
 {
 	return vm->config.userData;
