diff --git a/lib/libmicrohttpd/src/microhttpd/mhd_sockets.c b/lib/libmicrohttpd/src/microhttpd/mhd_sockets.c
index 47c2d9dc..27db6cc9 100644
--- a/lib/libmicrohttpd/src/microhttpd/mhd_sockets.c
+++ b/lib/libmicrohttpd/src/microhttpd/mhd_sockets.c
@@ -493,7 +493,7 @@ MHD_socket_create_listen_ (int use_ipv6)
                    0,
                    NULL,
                    0,
-                   WSA_FLAG_NO_HANDLE_INHERIT);
+                   WSA_FLAG_OVERLAPPED | WSA_FLAG_NO_HANDLE_INHERIT);
   cloexec_set = !0;
 #else  /* !SOCK_CLOEXEC */
   fd = MHD_INVALID_SOCKET;
