diff --git a/lib/wren/src/include/wren.h b/lib/wren/src/include/wren.h
index 583bbd20..23e34231 100644
--- a/lib/wren/src/include/wren.h
+++ b/lib/wren/src/include/wren.h
@@ -265,6 +265,7 @@ typedef enum
   WREN_TYPE_LIST,
   WREN_TYPE_NULL,
   WREN_TYPE_STRING,
+  WREN_TYPE_CLOSURE,
 
   // The object is of a type that isn't accessible by the C API.
   WREN_TYPE_UNKNOWN
diff --git a/lib/wren/src/vm/wren_vm.c b/lib/wren/src/vm/wren_vm.c
index 0e0101e6..9f01fe00 100644
--- a/lib/wren/src/vm/wren_vm.c
+++ b/lib/wren/src/vm/wren_vm.c
@@ -1579,6 +1579,7 @@ WrenType wrenGetSlotType(WrenVM* vm, int slot)
   if (IS_LIST(vm->apiStack[slot])) return WREN_TYPE_LIST;
   if (IS_NULL(vm->apiStack[slot])) return WREN_TYPE_NULL;
   if (IS_STRING(vm->apiStack[slot])) return WREN_TYPE_STRING;
+  if (IS_CLOSURE(vm->apiStack[slot])) return WREN_TYPE_CLOSURE;
   
   return WREN_TYPE_UNKNOWN;
 }
