diff --git a/lib/wren/src/vm/wren_vm.c b/lib/wren/src/vm/wren_vm.c
index 03b24e55..c9f6fee8 100644
--- a/lib/wren/src/vm/wren_vm.c
+++ b/lib/wren/src/vm/wren_vm.c
@@ -773,6 +773,11 @@ static Value getModuleVariable(WrenVM* vm, ObjModule* module,
 // also, as you can imagine, highly performance critical.
 static WrenInterpretResult runInterpreter(WrenVM* vm, register ObjFiber* fiber)
 {
+  if (vm->fiber != fiber)
+  {
+    vm->apiStack = NULL;
+  }
+
   // Remember the current fiber so we can find it if a GC happens.
   vm->fiber = fiber;
   fiber->state = FIBER_ROOT;
